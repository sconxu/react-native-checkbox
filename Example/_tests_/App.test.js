import React from 'react';
import findById from '../test/findById';
import renderer from 'react-test-renderer';
import App from '../App';

test('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

test('onChange of Checkbox true', () => {
  let checkBox = renderer.create(<App />).getInstance();
  checkBox.checkBoxState(true)
  expect(checkBox.state.checkbox).toEqual(false);
});

test('onChange of Checkbox false', () => {
  let checkBox = renderer.create(<App />).getInstance();
  checkBox.checkBoxState(false)
  expect(checkBox.state.checkbox).toEqual(true);
});

test('Checkbox exist', () => {
  const component = renderer.create(<App />).toJSON();
  expect(findById(component, 'checkBBox1')).toBeDefined();
  expect(findById(component, 'checkBBox2')).toBeDefined()
  expect(findById(component, 'checkBBox3')).toBeDefined()
  expect(findById(component, 'checkBBox4')).toBeDefined()
});