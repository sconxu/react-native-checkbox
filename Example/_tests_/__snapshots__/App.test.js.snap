// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#000",
        "borderBottomWidth": 5,
        "marginBottom": 10,
        "marginLeft": 45,
        "marginRight": 45,
        "marginTop": 10,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 8,
        }
      }
    >
      Note - Default Checkbox as describbed in README.md
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID="checkBBox1"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 26,
                "width": 26,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            label
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#000",
        "borderBottomWidth": 5,
        "marginBottom": 10,
        "marginLeft": 45,
        "marginRight": 45,
        "marginTop": 10,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 8,
        }
      }
    >
      Note - Checkbox with different Font size or Color of Label
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID="checkBBox2"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 26,
                "width": 26,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 15,
                },
                Object {
                  "color": "#FF0000",
                  "fontSize": 20,
                },
              ]
            }
          >
            CheckBox1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#000",
        "borderBottomWidth": 5,
        "marginBottom": 10,
        "marginLeft": 45,
        "marginRight": 45,
        "marginTop": 10,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 8,
        }
      }
    >
      Note - Checkbox with different Images
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID="checkBBox3"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 26,
                "width": 26,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            CheckBox1
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#000",
        "borderBottomWidth": 5,
        "marginBottom": 10,
        "marginLeft": 45,
        "marginRight": 45,
        "marginTop": 10,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 8,
        }
      }
    >
      Note - Checkbox with underlayColor Color
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID="checkBBox4"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 26,
                "width": 26,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            CheckBox1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
